main.py

```diff
@@ -1033,17 +1035,33 @@ def notify_interface_link_status(self, interface, reason):
 
     def notify_link_status_change(self, link, reason='not given'):
         """Send an event to notify about a status change on a link."""
-        name = 'kytos/topology.'
-        if link.status == EntityStatus.UP:
-            status = 'link_up'
+        link_id = link.id
+        if (
+            (not link.status_reason or link.status == EntityStatus.UP)
```

Comment:
Neat, @Ktmi. My only final concern here is multi-threaded access. To ensure that only a single thread will be able to enter in these conditionals at a time. On `topology` we've also been using `_links_notify_lock[link.id]` to notify only once in certain places, but trying to reaquire would lead to issues, so either it could be reused/restructure it or maybe you can introduce a new lock "companion" to protect `self.link_up` or something else equivalent.